app/main/api/handlers/encryptWallet.ts

```diff
@@ -38,13 +37,9 @@ export default async function encryptWallet(system: AppStateS & WalletStorageS,
   Promise<JsonSerializable> {
 
   return parseParams(params)
-    .then(async (p) => Promise.all([
-      Promise.resolve(p)
-        .then(inject(newWalletStorage, system.storageFactory)),
-      Promise.resolve(p)
-        .then(inject(getUnconsolidatedData, system.appStateManager))
-        .then(inject(newWallet, system.appStateManager))
-    ]))
+    .then(inject(getUnconsolidatedData, system.appStateManager))
+    .then(newWallet)
+    .then(inject(newWalletStorage, system.storageFactory))
     .then(storeWallet)
     .then(inject(replaceWallet, system))
```

Comment:
In #349 I've also mentioned some issues regarding the error handling in the promise chain. I think it is quite important that in the case of error at some point in this chain, we revert whichever actions need to be reverted. For instance, if there is an error in this line `.then(inject(replaceWallet, system))`, then we need to revert the actions performed at the previous steps. For the time being, I think that the only step that would need some revert action would be the storeWallet. However, when we add the storage of the WalletInfo in the plain storage, more 'undoing' actions will be required.